sdk/samples/router/general.py

```diff
@@ -0,0 +1,265 @@
+        resource_request=ResourceRequest(
+            min_replica=1,
+            max_replica=3,
+            cpu_request="500m",
+            memory_request="512Mi"
+        ),
+        endpoint="/echo",
+        timeout="60ms",
+        port=8080,
+        env=[],
+    )
+
+    # # Create the RouterConfig instance
+    router_config = RouterConfig(
+        environment_name="id-dev",
+        name="what-does-the-fox-say",
+        routes=routes,
+        rules=rules,
+        default_route_id="test",
+        experiment_engine=experiment_config,
+        resource_request=resource_request,
+        timeout="100ms",
+        log_config=log_config,
+        enricher=enricher,
+        ensembler=ensembler
+    )
+
+    # 1. Create a new router using the RouterConfig object
+    new_router = turing.Router.create(router_config)
+
+    # 2. List all routers
+    routers = turing.Router.list()
+    for r in routers:
+        if r.name == new_router.name:
+            my_router_id = r.id
+        print(r)
+
+    # 3. Get the router you just created using the router_id obtained
+    my_router = turing.Router.get(my_router_id)
+
+    # Access the router config from the returned Router object directly
+    my_router_config = my_router.config
+
+    # Modify something in the router config
+    my_router_config.routes.append(
+        Route(
+            id='fee-fi-fo-fum',
+            endpoint='http://fox-says.fee-fi-fo-fum',
+            timeout='20ms'
+        )
+    )
+
+    # 4. Update the router with the new router config
+    my_router.update(my_router_config)
+
+    # 5. List all the router config versions of your router
+    my_router_versions = my_router.list_versions()
+    for ver in my_router_versions:
+        if ver.status == 'undeployed':
+            first_ver_no = ver.version
+        if ver.status == "deployed":
+            latest_ver_no = ver.version
+        print(ver)
+
+    # 6. Deploy a specific router config version (the first one we created)
+    my_router.deploy_version(first_ver_no)
+
+    # 7. Undeploy the current active router configuration
+    my_router.undeploy()
+
+    # 8. Deploy the router's *current* configuration (notice how it still deploys the *first* version)
+    my_router.deploy()
+
+    # # Undeploy the router
+    my_router.undeploy()
+
+    # 9. Get a specific router version of the router
+    my_router_ver = my_router.get_version(first_ver_no)
+    print(my_router_ver)
+
+    # 10. Delete a specific router version of the router
+    my_router.delete_version(latest_ver_no)
+
+    # 11. Get all deployment events associated with this router
+    events = my_router.get_events()
+    for e in events:
+        print(e)
+
+    # 12. Delete this router (using its router_id)
+    turing.Router.delete(my_router_id)
+    # # Check if the router still exists
+    for r in turing.Router.list():
+        if r.name == my_router_config.name:
+            print("Oh my, this router still exists!")
```

Comment:
It'll be neater to use logging package instead of print statements.

Reply:
Hmm I was thinking using a basic `print` would be simpler for a sample file that's only meant to be read by users as a rough guide, but okay, using `logging` is fine too :)

Reply:
I think for sample this would suffice. Only if it's part of the core SDK then I would suggest using a logger so that the user can disable or set the levels on the logger.